src/fidesops/ops/service/connectors/saas/connector_registry_service.py

```diff
@@ -133,3 +140,129 @@ def load_registry(config_file: str) -> ConnectorRegistry:
     if _registry is None:
         _registry = ConnectorRegistry.parse_obj(load_toml([config_file]))
     return _registry
+
+
+def update_saas_configs(registry: ConnectorRegistry, db: Session) -> None:
+    """
+    Updates SaaS config instances currently in the DB if to the
+    corresponding template in the registry are found.
+
+    Effectively an "update script" for SaaS config instances,
+    to be run on server bootstrap.
+    """
+    for connector_type in registry.connector_types():
+        logger.debug(
+            "Determining if any updates are needed for connectors of type %s based on templates...",
+            connector_type,
+        )
+        template: ConnectorTemplate = registry.get_connector_template(  # type: ignore
+            connector_type
+        )
+        saas_config_template = SaaSConfig.parse_obj(load_config(template.config))
+        template_version: Union[LegacyVersion, Version] = parse_version(
+            saas_config_template.version
+        )
+
+        connection_configs: Iterable[ConnectionConfig] = ConnectionConfig.filter(
+            db=db,
+            conditions=(ConnectionConfig.saas_config["type"].astext == connector_type),
+        ).all()
+        for connection_config in connection_configs:
+            saas_config_instance = SaaSConfig.parse_obj(connection_config.saas_config)
+            if parse_version(saas_config_instance.version) < template_version:
+                logger.info(
+                    "Updating SaaS config instance '%s' of type '%s' as its version, %s, was found to be lower than the template version %s",
+                    saas_config_instance.fides_key,
+                    connector_type,
+                    saas_config_instance.version,
+                    template_version,
+                )
+                try:
+                    update_saas_instance(
+                        db,
+                        connection_config,
+                        template,
+                        saas_config_instance,
+                    )
+                except Exception:
+                    logger.error(
+                        "Encountered error attempting to update SaaS config instance %s",
+                        saas_config_instance.fides_key,
+                        exc_info=True,
+                    )
+
+
+def update_saas_instance(
```

Comment:
I _would_ like to see an end-to-end test of this method.

Reply:
Clarified in person, Adam's existing test harness is bigger picture than this and it is being covered in a more end-to-end way than i realized.